[chimera-pverify] Review

wschoenell opened the pull request:
I am going thru pverify and fixing all the bugs that I find and make it usable again. This version is a working version for start. A LOT should be improved, but it at least works now.

The main changes are:

* Added a `download` method to `ImageUtil` for downloading remote images. This is useful when you have the camera running on another computer than the one where you are running the pverify;

* Removed IRAF dependency;

* A lot of code organization, removed not-used imports, changed `print()`s by `log.debug()`s etc on (and only) pverify methods.

ankanaan commented:
pverify is and always perfect :-)

On Fri, Jul 31, 2015 at 12:51 PM, William Schoenell <
notifications@github.com> wrote:

> I am going thru pverify and fixing all the bugs that I find and make it
> usable again. This version is a working version for start. A LOT should be
> improved, but it at least works now.
>
> The main changes are:
>
>    -
>
>    Added a download method to ImageUtil for downloading remote images.
>    This is useful when you have the camera running on another computer than
>    the one where you are running the pverify;
>    -
>
>    Removed IRAF dependency;
>    -
>
>    A lot of code organization, removed not-used imports, changed print()s
>    by log.debug()s etc on (and only) pverify methods.
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/astroufsc/chimera/pull/99
> Commit Summary
>
>    - Added util to automatically download images from http server
>    - Many bug fixes on chimera-pverify.
>
> File Changes
>
>    - *M* setup.py
>    <https://github.com/astroufsc/chimera/pull/99/files#diff-0> (5)
>    - *M* src/chimera/controllers/pointverify.py
>    <https://github.com/astroufsc/chimera/pull/99/files#diff-1> (112)
>    - *M* src/chimera/util/astrometrynet.py
>    <https://github.com/astroufsc/chimera/pull/99/files#diff-2> (796)
>    - *M* src/chimera/util/image.py
>    <https://github.com/astroufsc/chimera/pull/99/files#diff-3> (30)
>    - *M* src/scripts/chimera-pverify
>    <https://github.com/astroufsc/chimera/pull/99/files#diff-4> (33)
>
> Patch Links:
>
>    - https://github.com/astroufsc/chimera/pull/99.patch
>    - https://github.com/astroufsc/chimera/pull/99.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/astroufsc/chimera/pull/99>.
>

phsilva commented on setup.py:
```diff
@@ -70,7 +67,7 @@
 
     # installation happens in the specified order
-    install_requires=[
+    install_requires=[  "astropy",
```
duplicated? astropy is already included.

wschoenell replied:
I was betrayed by PyCharm!

phsilva commented on src/chimera/controllers/pointverify.py:
```diff
@@ -127,39 +125,30 @@ def pointVerify(self):
-            (wcs["NAXIS2"] / 2.0 - wcs["CRPIX2"]) * wcs["CD2_2"]
-        currentWCS = Position.fromRaDec(Coord.fromD(ra_wcs_center),
-                                        Coord.fromD(dec_wcs_center))
+        wcs = fits.getheader(wcs_name)
```
maybe put this in a method on Image class? looks like something useful.

wschoenell replied:
Done in 03fed072fd09f3c2665d95d7de5ad9a5a494ddb3

wschoenell commented:
Added more changes:

* Change the filter to `filter` config directive file when exposing;
* Documented `pverify` interface. Changed `tolra` and `toldec` config parameters from `max_tolerance` which is equals to `sqrt(delta_ra**2 + delta_dec**2)`, `max_trials` is `max_tries` now for compatibility with the other interfaces;
* Fixed an issue of determining files base name when remote client is windows and local is linux and vice-versa;
* Added `worldAtCenter()` method to `Image` class;

ankanaan commented:
On Mon, Aug 3, 2015 at 10:26 AM, William Schoenell <notifications@github.com
> wrote:

> Added more changes:
>
>    - Change the filter to filter config directive file when exposing;
>    - Documented pverify interface. Changed tolra and toldec config
>    parameters from max_tolerance which is equals to sqrt(delta_ra**2 +
>    delta_dec**2), max_trials is max_tries now for compatibility with the
>    other interfaces;
>
>
not sure I like this.  Telescopes often point better in one direction, in
which case it makes sense to have different tolerances - at least for
equatorial mounts.


>
>    - Fixed an issue of determining files base name when remote client is
>    windows and local is linux and vice-versa;
>    - Added worldAtCenter() method to Image class;
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/astroufsc/chimera/pull/99#issuecomment-127233150>.
>

wschoenell commented:
@ankanaan,
Yes. I see your point. I just changed back to what it was on 5203d3b.

Thanks!

wschoenell commented:
Merging this to afterwards move into a plugin
